static/app/components/events/meta/annotatedText/index.tsx

```diff
@@ -1,123 +1,32 @@
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+import {useRouteContext} from 'sentry/utils/useRouteContext';
 
-import List from 'sentry/components/list';
-import ListItem from 'sentry/components/list/listItem';
-import Tooltip from 'sentry/components/tooltip';
-import {IconWarning} from 'sentry/icons';
-import {t} from 'sentry/locale';
-import space from 'sentry/styles/space';
-import {MetaError} from 'sentry/types';
-
-import Chunks from './chunks';
-import {getTooltipText} from './utils';
-import ValueElement from './valueElement';
+import {AnnotatedTextErrors} from './annotatedTextErrors';
+import {AnnotatedTextValue} from './annotatedTextValue';
 
 type Props = {
   value: React.ReactNode;
   className?: string;
   meta?: Record<any, any>;
 };
 
-const AnnotatedText = ({value, meta, className, ...props}: Props) => {
-  const renderValue = () => {
-    if (meta?.chunks?.length && meta.chunks.length > 1) {
-      return <Chunks chunks={meta.chunks} />;
-    }
-
-    if (meta?.rem?.length) {
-      return (
-        <Tooltip
-          title={getTooltipText({rule_id: meta.rem[0][0], remark: meta.rem[0][1]})}
-        >
-          <ValueElement value={value} meta={meta} />
-        </Tooltip>
-      );
-    }
-
-    return <ValueElement value={value} meta={meta} />;
-  };
-
-  const formatErrorKind = (kind: string) => {
-    return capitalize(kind.replace(/_/g, ' '));
-  };
-
-  const getErrorMessage = (error: MetaError) => {
-    const errorMessage: string[] = [];
-
-    if (Array.isArray(error)) {
-      if (error[0]) {
-        errorMessage.push(formatErrorKind(error[0]));
-      }
-
-      if (error[1]?.reason) {
-        errorMessage.push(`(${error[1].reason})`);
-      }
-    } else {
-      errorMessage.push(formatErrorKind(error));
-    }
-
-    return errorMessage.join(' ');
-  };
-
-  const getTooltipTitle = (errors: Array<MetaError>) => {
-    if (errors.length === 1) {
-      return <TooltipTitle>{t('Error: %s', getErrorMessage(errors[0]))}</TooltipTitle>;
-    }
-    return (
-      <TooltipTitle>
-        <span>{t('Errors:')}</span>
-        <StyledList symbol="bullet">
-          {errors.map((error, index) => (
-            <ListItem key={index}>{getErrorMessage(error)}</ListItem>
-          ))}
-        </StyledList>
-      </TooltipTitle>
-    );
-  };
-
-  const renderErrors = (errors: Array<MetaError>) => {
-    if (!errors.length) {
-      return null;
-    }
-
-    return (
-      <StyledTooltipError title={getTooltipTitle(errors)}>
-        <StyledIconWarning color="red300" />
-      </StyledTooltipError>
-    );
-  };
+export function AnnotatedText({value, meta, className, ...props}: Props) {
+  const organization = useOrganization();
+  const routerContext = useRouteContext();
+  const projectId = routerContext.location.query.project;
+  const {projects} = useProjects();
+  const currentProject = projects.find(project => project.id === projectId);
```

Comment:
Are we guaranteed that the correct project will be in the store? Let's say if org has hundreds of projects.

Reply:
hm a good point... I will double-check it and maybe address this in a follow-up. In any case, the tooltip with PII has a fall back logic for this case